build(aio): fail build if more than h1 is found in a doc

petebacondarwin opened the pull request:
Closes #16193

petebacondarwin committed:
docs(aio): remove excess h1s from guides

Only one h1 is allowed per document.

(Also took the opportunity to remove unnecessary blank lines from these
docs, and a bit of general tidying.)

Closes #16193

mary-poppins commented:
The angular.io preview for e6296db04b17ada26bed9110f8081f0e3bd06732 is available [here][1].

[1]: https://pr16679-e6296db04b17ada26bed9110f8081f0e3bd06732.ngbuilds.io/

mary-poppins commented:
The angular.io preview for b5cc53f47859740f2cf57aceb5a3b8285eab80f9 is available [here][1].

[1]: https://pr16679-b5cc53f47859740f2cf57aceb5a3b8285eab80f9.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 036fe972d3498e232ba9fd270439324763ef6948 is available [here][1].

[1]: https://pr16679-036fe972d3498e232ba9fd270439324763ef6948.ngbuilds.io/

gkalpak commented on packages/common/src/directives/ng_if.ts:
```diff
@@ -32,15 +32,15 @@ import {Directive, EmbeddedViewRef, Input, TemplateRef, ViewContainerRef} from '
  * a binding (just like `else`). Because `then` and `else` are bindings, the template references can
  * change at runtime as shown in this example.
- *
+ *ß
```
??

petebacondarwin replied:
weird!!

gkalpak commented on aio/tools/transforms/post-process-package/processors/post-process-html.js:
```diff
@@ -28,19 +28,20 @@ module.exports = function postProcessHtml(log, createDocMessage) {
-          const vFile = engine.processSync(doc.renderedContent);
-          vFile.messages.forEach(m => {
-            const message = createDocMessage(m.message, doc);
-            if (m.fatal) {
```
What does this mean for `fatal` errors? Are there any? Are we ignoring them (just logging)?

petebacondarwin replied:
So if you use `vFile.fatal()` to create a fatal error message, it immediately throws. So this code was never being used called. It is possible to create a message and manually set its fatal property, in which case we would only log it, along with warning messages.

gkalpak commented on aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js:
```diff
@@ -0,0 +1,28 @@
+    `
+    };
+    expect(() => processor.$process([doc])).toThrowError(createDocMessage('More than one h1 found [<h1>Heading 1, <h1>Heading 1a</h1>]', doc));
+  });
```
What happens with 0 or 1 h1s? Where's the test 😛

petebacondarwin replied:
OK OK

petebacondarwin committed:
build(aio): fail build if more than h1 is found in a doc

Closes #16193

petebacondarwin committed:
docs(common): remove h1s from API docs

Only one h1 is allowed per document, and this is provided by the template.
So we cannot have any h1 tags (or `#` markdown shorthand) in any API docs.

Closes #16193

petebacondarwin commented:
@gkalpak PTAL

mary-poppins commented:
The angular.io preview for f35370e03d3c9d187e6deb8873c357c19af4a11b is available [here][1].

[1]: https://pr16679-f35370e03d3c9d187e6deb8873c357c19af4a11b.ngbuilds.io/

gkalpak commented on aio/tools/transforms/angular-base-package/post-processors/h1-checker.spec.js:
```diff
@@ -0,0 +1,49 @@
+      docType: 'a',
+      renderedContent: `
+        <h2>Heading 2</h2>
+    `
```
Nit: Uncommon indentation of closing `` ` ``.

petebacondarwin replied:
Maybe but it appears elsewhere in the transforms stuff, so...

mary-poppins commented:
The angular.io preview for 093743a6ef3cf3ed0471c8f827df4154b4fec1f3 is available [here][1].

[1]: https://pr16679-093743a6ef3cf3ed0471c8f827df4154b4fec1f3.ngbuilds.io/
